inc/plugin-logic/class-admin-handler.php

```diff
@@ -155,4 +160,75 @@ public function include_choctaw_events_post_type_in_search( \WP_Query $query ) {
 			$query->set( 'post_type', array( 'choctaw-events' ) );
 		}
 	}
+
+	/** Create a cron job to expire events */
+	protected function schedule_event_expiry() {
+
+		if ( ! wp_next_scheduled( 'expire_choctaw_event_posts' ) ) {
+			wp_schedule_event( time(), 'hourly', 'expire_choctaw_event_posts' );
+		}
+
+		add_action( 'expire_choctaw_event_posts', array( $this, 'expire_choctaw_events' ), 100 );
+	}
+
+	/**
+	 * Compares today to the ACF date/time fields and updates the post status to "draft" if the event is expired */
+	public function expire_choctaw_events() {
+		$timezone = new DateTimeZone( 'America/Chicago' );
+		$today    = new DateTime( 'now', $timezone );
+		$events   = get_posts(
+			array(
+				'post_type'      => array( 'choctaw-events' ),
+				'posts_per_page' => -1,
+			)
+		);
+
+		foreach ( $events as $event ) {
+			$event_details    = get_field( 'event_details', $event->ID );
+			$event_is_expired = $this->get_event_expiry( $event_details['time_and_date'] ) < $today;
+
+			if ( $event_is_expired ) {
+				$postdata = array(
+					'ID'          => $event->ID,
+					'post_status' => 'draft',
+				);
+				wp_update_post( $postdata );
+			}
+		}
+	}
+
+	/**
+	 * Get the event expiry date
+	 *
+	 * @param array $time_and_date the time and date
+	 * @return DateTime
+	 */
+	private function get_event_expiry( array $time_and_date ): DateTime {
+		$timezone = new DateTimeZone( 'America/Chicago' );
+		if ( $time_and_date['is_all_day'] ) {
+			$expiry = new \DateTime( $time_and_date['start_date'], $timezone );
+			$expiry->modify( '+1 day' );
+		} else {
+			if ( empty( $time_and_date['end_date'] ) ) {
+				$time_and_date['end_date'] = $time_and_date['start_date'];
+			}
+			if ( empty( $time_and_date['end_time'] ) ) {
+				$time_and_date['end_time'] = date( 'h:i A', strtotime( $time_and_date['start_time'] . ' +1 hour' ) ); // phpcs:ignore
```

Comment:
This says if end_time isn't set to set it equal to the start_time + 1 hour, but I don't believe that's correct.

If there isn't an end_time, it should be set to the end of the end_date (you can use 11:59pm, if that's easier) UNLESS there also isn't an end_date, then it should be set to whatever the start_time is +1 hour UNLESS there also isn't a start_time, which then it should be set to the end of the start_date.

Reply:
That's a good catch. I forgot how weird some of this event logic is for markup purposes. Got it sorted.